src/vbms/client.rb

```
@@ -142,23 +142,44 @@ def build_request(body, headers)
     # rubocop:enable Metrics/AbcSize
 
     def process_response(request, response)
-      soap = response.body.match(%r{<soap:envelope.*?</soap:envelope>}im)[0]
-      doc = Nokogiri::XML(soap)
+      # we could check the response content-type to make sure it's XML, but they don't seem
+      # to send any HTTP headers back, so we'll just have to do a quick check to see if it
+      # looks like XML and then go from there
+      unless response.body.match(%r{<\?xml version})
+        fail SOAPError, "Request did not return XML"
+      end
+
+      begin
+        full_doc = Nokogiri::XML(response.body)
+      rescue
```

I generally try to catch the exception only if I'm going to do something useful with it; here there's no logging or anything else to be done. Wrapping the error doesn't seem to decrease the error surface of the API, so I'd be prone to let the exception bubble up here. (Though totally OK with wrapping if it simplifies things in some way)

There should definitely should be some tests to show what happens when each of these exception clauses is triggered; right now I think only the exception at line 161 is triggered by the tests, not this one or line 178.